src/components/Button/Button.jsx

```diff
@@ -86,17 +82,19 @@ const Button = React.createClass({
 			isDisabled,
 			onClick,
 		} = this.props;
+		const domNode = ReactDOM.findDOMNode(this);
 
 		if (!isDisabled) {
 			onClick({ event, props: this.props });
+			// required to correctly apply the focus state in Safari and Firefox
+			domNode.focus();
```

Comment:
i'm cool with it, but i think it should be before the `onClick({...})` call, that way the last thing that happens in this function is passing control on to a handler.